app/PaymentDrivers/CheckoutComPaymentDriver.php

```diff
@@ -0,0 +1,271 @@
+    {
+        if ($gateway_type_id == GatewayType::CREDIT_CARD) {
+            return 'gateways.checkout.credit_card';
+        }
+
+        if ($gateway_type_id == GatewayType::TOKEN) {
+            return 'gateways.checkout.credit_card';
+        }
+    }
+
+    public function processPaymentView(array $data)
+    {
+        $data['gateway'] = $this;
+        $data['client'] = $this->client;
+        $data['currency'] = $this->client->getCurrencyCode();
+        $data['value'] = $this->convertToCheckoutAmount($data['amount_with_fee'], $this->client->getCurrencyCode());
+        $data['raw_value'] = $data['amount_with_fee'];
+        $data['customer_email'] = $this->client->present()->email;
+
+        return render($this->viewForType($data['payment_method_id']), $data);
+    }
+
+    public function processPaymentResponse($request)
+    {
+        $this->init();
+
+        $state = [
+            'server_response' => json_decode($request->gateway_response),
+            'value' => $request->value,
+            'raw_value' => $request->raw_value,
+            'currency' => $request->currency,
+        ];
+
+        $state = array_merge($state, $request->all());
+
+        $method = new TokenSource($state['server_response']->cardToken);
+
+        $payment = new Payment($method, $state['currency']);
+        $payment->amount = $state['value'];
+
+        // We need a proper logic to test if this is first payment with Checkout.
+        // Pseudo: if ($this->client->payments()->where('gateway_id', PaymentType::TYPE_CHECKOUT)->count() == 0)
+
+        $first_payment = true;
+
+        if ($first_payment && $this->client->currency()->code === 'EUR') {
+            $payment->{"3ds"} = [
+                'enabled' => true,
+            ];
+        }
+
+        try {
+            $response = $this->gateway->payments()->request($payment);
+            $state['payment_response'] = $response;
+
+            if ($response->status === 'Authorized') {
+                return $this->processSuccessfulPayment($state);
+            }
+
+            if ($response->status === 'Pending') {
+                return $this->processPendingPayment($state);
+            }
+
+            if ($response->status === 'Declined') {
+                return $this->processUnsuccessfulPayment($state);
+            }
+        } catch (CheckoutHttpException $e) {
+            return $this->processInternallyFailedPayment($e, $state);
+        }
+    }
+
+    public function processSuccessfulPayment($state)
+    {
+        $state['charge_id'] = $state['payment_response']->id;
+
+        if (isset($state['store_card'])) {
+            $this->saveCard($state);
+        }
+
+        $data = [
+            'payment_method' => $state['charge_id'],
+            'payment_type' => PaymentType::CREDIT_CARD_OTHER, // @todo: needs proper status
+            'amount' => $state['raw_value'],
+        ];
+
+        $payment = $this->createPayment($data, NinjaPaymentModel::STATUS_COMPLETED);
```

Comment:
The status should use the native Payment:: status... in this case Payment::STATUS_COMPLETED